File: app/src/main/java/com/example/proyecto_final/MihogarActivity.java
package com.example.proyecto_final;

import android.annotation.SuppressLint;
import android.content.Intent;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.QueryDocumentSnapshot;
import com.google.firebase.firestore.QuerySnapshot;

public class MihogarActivity extends AppCompatActivity {

    Button btnEditar;
    ImageView btnMascota;

    ImageButton btnAtras;


    FirebaseAuth auth = FirebaseAuth.getInstance();

    // Obtiene el usuario actualmente conectado
    FirebaseUser currentUser = auth.getCurrentUser();



    FirebaseFirestore db = FirebaseFirestore.getInstance();


    private DatabaseReference mDatabase;



    @SuppressLint("MissingInflatedId")
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_mihogar);



        mDatabase = FirebaseDatabase.getInstance().getReference();


        // Si el usuario está conectado, muestra su información
        if (currentUser != null) {
            String email = currentUser.getEmail();


            // Actualiza la interfaz de usuario con la información del usuario
            // ...


            db.collection("Perfiles")
                    .get()
                    .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
                        @Override
                        public void onComplete(@NonNull Task<QuerySnapshot> task) {
                            if (task.isSuccessful()) {
                                for (QueryDocumentSnapshot document : task.getResult()) {
                                    Log.d("Seccessful:", document.getId() + " => " + document.getData());

                                    if (email.equals(document.getData().get("email-perfil"))) {
                                        TextView textView = findViewById(R.id.user_name);

                                        textView.setText((CharSequence) document.getData().get("nombre_perfil"));
                                    }
                                }
                            } else {
                                Log.w("Error getting documents.", task.getException());
                            }
                        }
                    });
        }







        btnAtras = findViewById(R.id.backButton);

        btnAtras.setOnClickListener(new View.OnClickListener() { // Boton Editar
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MihogarActivity.this, HomeActivity.class);
                startActivity(intent);
                finish();
            }
        });


        btnEditar = findViewById(R.id.botonEditar);

        btnEditar.setOnClickListener(new View.OnClickListener() { // Boton Editar
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MihogarActivity.this, CrearPerfilActivity.class);
                startActivity(intent);
            }
        });

        btnMascota = findViewById(R.id.mascotaAddBoton);

        btnMascota.setOnClickListener(new View.OnClickListener() { // Boton Agregar Mascota
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MihogarActivity.this, MascotasActivity.class);
                startActivity(intent);
                finish();
            }
        });

    }
}

File: app/src/main/java/com/example/proyecto_final/MihogarActivity.kt
package com.example.proyecto_final

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.firestore.FirebaseFirestore

class MihogarActivity : AppCompatActivity() {
    var btnEditar: Button? = null
    var btnMascota: ImageView? = null
    var btnAtras: ImageButton? = null
    var auth = FirebaseAuth.getInstance()

    // Obtiene el usuario actualmente conectado
    var currentUser = auth.currentUser
    var db = FirebaseFirestore.getInstance()
    private var mDatabase: DatabaseReference? = null
    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_mihogar)
        mDatabase = FirebaseDatabase.getInstance().reference


        // Si el usuario está conectado, muestra su información
        if (currentUser != null) {
            val email = currentUser!!.email


            // Actualiza la interfaz de usuario con la información del usuario
            // ...
            db.collection("Perfiles")
                .get()
                .addOnCompleteListener { task ->
                    if (task.isSuccessful) {
                        for (document in task.result) {
                            Log.d("Seccessful:", document.id + " => " + document.data)
                            if (email == document.data["email-perfil"]) {
                                val textView = findViewById<TextView>(R.id.user_name)
                                textView.text = document.data["nombre_perfil"] as CharSequence?
                            }
                        }
                    } else {
                        Log.w("Error getting documents.", task.exception)
                    }
                }
        }
        btnAtras = findViewById(R.id.backButton)
        btnAtras.setOnClickListener(
            View.OnClickListener
            // Boton Editar
            {
                val intent = Intent(this@MihogarActivity, HomeActivity::class.java)
                startActivity(intent)
                finish()
            })
        btnEditar = findViewById(R.id.botonEditar)
        btnEditar.setOnClickListener(
            View.OnClickListener
            // Boton Editar
            {
                val intent = Intent(this@MihogarActivity, CrearPerfilActivity::class.java)
                startActivity(intent)
            })
        btnMascota = findViewById(R.id.mascotaAddBoton)
        btnMascota.setOnClickListener(
            View.OnClickListener
            // Boton Agregar Mascota
            {
                val intent = Intent(this@MihogarActivity, MascotasActivity::class.java)
                startActivity(intent)
                finish()
            })
    }
}